From d7e318db6bb10637996c695bae6cd52e648a3ee1 Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Mon, 22 Apr 2024 15:27:12 +0200
Subject: [PATCH] imx8mn-compact: update for u-boot ethernet phy detection

Boards version 1.2 and later have analog devices phy at address 0,
earlier versions have atheros phy at address 4.

To correctly support old boards on old u-boot, link by default
phy-handle toi ethernet-phy@4 and keep the node enabled.

New boards should use new u-boot with runtime-patching of DTB.
Disable the ethernet-phy@0 node by default, u-boot will enable it, and
patch the phy-handle during boot.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 arch/arm64/boot/dts/freescale/imx8mn-compact.dts | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/freescale/imx8mn-compact.dts b/arch/arm64/boot/dts/freescale/imx8mn-compact.dts
index 465ff1f358a0..1d32fe9068c0 100755
--- a/arch/arm64/boot/dts/freescale/imx8mn-compact.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mn-compact.dts
@@ -572,6 +572,7 @@ &fec1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_fec1>;
 	phy-mode = "rgmii-id";
+	phy-handle = <&ethphy4>;
 	phy-reset-gpios = <&gpio3 19 GPIO_ACTIVE_LOW>;
 	fsl,magic-packet;
 	status = "okay";
@@ -593,7 +594,7 @@ ethphy0: ethernet-phy@0 {
 			reg = <0>;
 			at803x,led-act-blind-workaround;
 			at803x,eee-disabled;
-			status = "okay";
+			status = "disabled";
 		};
 	};
 };
-- 
2.35.3

